server/src/main/java/org/elasticsearch/search/crossproject/CrossProjectIndexResolutionValidator.java

```diff
@@ -138,33 +167,82 @@ public static ElasticsearchException validate(
                 // checking if flat expression matched remotely
                 for (String remoteExpression : remoteExpressions) {
                     String[] splitResource = splitQualifiedResource(remoteExpression);
-                    ElasticsearchException exception = checkSingleRemoteExpression(
+                    var projectAlias = splitResource[0];
+                    var resource = splitResource[1];
+
+                    ElasticsearchException remoteException = checkSingleRemoteExpression(
                         remoteResolvedExpressions,
-                        splitResource[0], // projectAlias
-                        splitResource[1], // resource
+                        projectAlias,
+                        resource,
                         remoteExpression,
                         indicesOptions
                     );
-                    if (exception == null) {
+                    if (remoteException == null) {
                         // found flat expression somewhere
                         foundFlat = true;
                         break;
                     }
-                    if (false == isUnauthorized && exception instanceof ElasticsearchSecurityException) {
+                    if (false == isUnauthorized && remoteException instanceof ElasticsearchSecurityException) {
                         isUnauthorized = true;
+                        if (remoteAuthorizationExceptions == null) {
+                            remoteAuthorizationExceptions = new HashMap<>();
+                            remoteUnauthorizedIndices = new HashMap<>();
+                        }
+                        remoteAuthorizationExceptions.putIfAbsent(projectAlias, remoteException);
+                        remoteUnauthorizedIndices.compute(projectAlias, (k, v) -> {
+                            if (v == null) {
+                                v = new ArrayList<>();
+                            }
+                            v.add(resource);
+                            return v;
+                        });
                     }
                 }
                 if (foundFlat) {
                     continue;
                 }
```

Comment:
It seems to me that we need to clear entries populated in this cycle for `remoteAuthorizationExceptions`. If we have 2 links, `p1` and p2` and `logs` is accessible only on `p2`. I think we'd populate `remoteAuthorizationExceptions` for `p1` and then `foundFlat` becomes `true` for `p2`. This should be a success case. But I think the code is going to report exception?

Reply:
How do we expect `ignoreUnavailable` to behave with flat expression authorization errors? At the moment, when it's true, any authorization errors are ignored, and when it's false, we have the behaviour you're describing